### ISSUE TITLE
Add counter variant to Input

### ISSUE THREAD

**Issue by interactivellama** [salesforce/design-system-react#1523]
See Counter variant https://www.lightningdesignsystem.com/components/input/

It's like Spinbox from Fuel UX

I gut is thinking that you have a `value`, `increment` prop and `onIncrement`, `onDecrement` event props.

It may be worth breaking out into it's own component. `onIncrement`, `onDecrement` event props seem odd on text input.

**Comment by kevinparkerson** [comment]
@interactivellama this is what I have planned for API:
```
/**
 * Specifies maximum accepted value for a counter input
 */
max: PropTypes.number,
/**
 * Specifies minimum accepted value for a counter input
 */
min: PropTypes.number,
/**
 * Determines the step size upon increment or decrement. Can be set to decimal values.
 */
step: PropTypes.number,
/**
 * Fired every time a counter input is decremented
 */
onDncrement: PropTypes.func,
/**
 * Fired every time a counter input is incremented
 */
onIncrement: PropTypes.func,
/**
 * The input is a controlled component, and will always display this value.
 */
value: PropTypes.oneOfType([
	PropTypes.number,
	PropTypes.string,
]),
```
Based on what seemed essential according to [MDN's entry on the counter input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number) as well as the important features in Fuel UX's spinbox.

**Comment by interactivellama** [comment]
Sounds good to me!

**Comment by interactivellama** [comment]
`value` should probably just be a string, since we are re-using `Input` unless we want to allow a number to passed into a normal `input`.

**Comment by interactivellama** [comment]
We could call a custom `checkProp` to error out if you pass a number into a text `input` as an option, too.

**Comment by kevinparkerson** [comment]
> We could call a custom `checkProp` to error out if you pass a number into a text `input` as an option, too.

I like this idea, though perhaps it should be a warning instead?

**Comment by interactivellama** [comment]
Yeah, there is low warning option. https://github.com/salesforce/design-system-react/blob/master/utilities/warning/low-priority-warning.js#L6
